library/VTKExtensions/Rendering/vtkF3DDropZoneActor.cxx

```diff
@@ -0,0 +1,131 @@
+#include <vtkViewport.h>
+
+vtkStandardNewMacro(vtkF3DDropZoneActor);
+
+//------------------------------------------------------------------------------
+vtkF3DDropZoneActor::vtkF3DDropZoneActor()
+{
+  this->TextActor->SetMapper(this->TextMapper);
+  vtkTextProperty* tProp = this->TextMapper->GetTextProperty();
+  tProp->SetFontSize(25);
+  tProp->SetJustificationToCentered();
+  tProp->SetVerticalJustificationToCentered();
+
+  this->BorderMapper->SetInputData(this->BorderPolyData);
+  this->BorderActor->SetMapper(this->BorderMapper);
+}
+
+//------------------------------------------------------------------------------
+vtkF3DDropZoneActor::~vtkF3DDropZoneActor() = default;
+
+//------------------------------------------------------------------------------
+void vtkF3DDropZoneActor::ReleaseGraphicsResources(vtkWindow* win)
+{
+  this->TextActor->ReleaseGraphicsResources(win);
+  this->BorderActor->ReleaseGraphicsResources(win);
+}
+
+//------------------------------------------------------------------------------
+int vtkF3DDropZoneActor::RenderOverlay(vtkViewport* viewport)
+{
+  const int* vSize = viewport->GetSize();
+
+  // Simply position text actor in the middle
+  this->TextMapper->SetInput(this->DropText.c_str());
+  this->TextActor->SetPosition(vSize[0] / 2, vSize[1] / 2);
+  this->TextActor->RenderOverlay(viewport);
+
+  if (this->BuildBorderGeometry(viewport))
+  {
+    this->BorderActor->RenderOverlay(viewport);
+  }
+  return 1;
+}
+
+//------------------------------------------------------------------------------
+bool vtkF3DDropZoneActor::BuildBorderGeometry(vtkViewport* viewport)
+{
+  constexpr int tickWidth = 3;
+  constexpr int tickLength = 10;
+
+  const int* vSize = viewport->GetSize();
+  if (vSize[0] == this->ComputedBorderViewportSize[0] &&
+    vSize[1] == this->ComputedBorderViewportSize[1])
+  {
+    return true;
+  }
+
+  // padding is 10% of shortest side
+  const int padding = std::min(vSize[0], vSize[1]) * 0.1;
+  const int borderW = vSize[0] - 2 * padding;
+  const int borderH = vSize[1] - 2 * padding;
+
+  // number of ticks assuming spacing == tickLength
+  const int nTicksW = borderW / (tickLength * 2);
+  const int nTicksH = borderH / (tickLength * 2);
+
+  // recompute uniform spacing
+  const double spacingW = static_cast<double>(borderW - nTicksW * tickLength) / (nTicksW - 1);
+  const double spacingH = static_cast<double>(borderH - nTicksH * tickLength) / (nTicksH - 1);
+
+  vtkNew<vtkPoints> borderPoints;
+  vtkNew<vtkCellArray> borderCells;
+  borderPoints->SetNumberOfPoints(4 * (2 * nTicksW + 2 * nTicksH));
+  vtkIdType index = 0;
+
+  // Draw top/bottom
+  for (int i = 0; i < nTicksW; ++i)
+  {
+    const int x0 = padding + tickLength * i + spacingW * i;
+    const int x1 = x0 + tickLength;
+    const int y[2] = { padding, vSize[1] - padding };
+    const int h[2] = { +tickWidth, -tickWidth };
+    for (int j = 0; j < 2; ++j)
+    {
+      vtkIdType ids[4] = { index++, index++, index++, index++ };
```

Comment:
could be `const`

Reply:
https://stackoverflow.com/questions/10747927/should-i-use-const-for-local-variables-for-better-code-optimization

actually, I removed many const

Reply:
The link you pasted recommends using const whenever it's possible 🤷

Reply:
However I guess it's causing issues with `InsertNextCell` later, right?

Reply:
```
You should rather use const for ensuring const-correctness and not for micro-optimization.
const correctness lets compiler help you guard against making honest mistakes, so you should use const wherever possible but for maintainability reasons & preventing yourself from doing stupid mistakes. 
```

to my understanding, it does not.,

Reply:
I will just merge this but I'm open to a discussion about the const usage in F3D if needed.